### PR TITLE
Completed component design pattern implementation, testing and respective README

### PR DESCRIPTION
**Pull request by LeseChe**
Completed component design pattern implementation, testing and respective README
Related issue: #556 

This is a completed implementation of the component design pattern, it is a simple implementation demonstrating the use of this pattern for two game objects with varying components. We have done simple tests of the code which has provided 100% code coverage of the implemented methods and there is a completed readme file with all the necessary info regarding this pattern.

Authors: @LeseChe @SammanPali

### CODE REVIEW

**Comment by SammanPali** [comment]
PR updates (25/10/22):

- uses Lombok to remove getter/setter boilerplate 
- updated grammar, readme and comments 
- replaced System prints with LOGGER as per repo standards

File: component/src/main/java/com/iluwatar/component/component/graphiccomponent/ObjectGraphicComponent.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+   */
+  @Override
+  public void update(GameObject gameObject) {
+    if (gameObject.getVelocity() > 0) {
+      LOGGER.info(gameObject.getName() + " has a positive velocity.");
+    } else if (gameObject.getVelocity() < 0) {
+      LOGGER.info(gameObject.getName() + " has a negative velocity.");
+    } else {
+      LOGGER.info(gameObject.getName() + " has zero velocity.");
+    }
```
Comment (iluwatar) [inline component/src/main/java/com/iluwatar/component/component/graphiccomponent/ObjectGraphicComponent.java:25]: Hmm maybe just log the current velocity?

File: component/src/test/java/com/iluwatar/component/GameObjectTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    GameObject npcTest = GameObject.createNpc();
+
+    //note that velocity and coordinates are initialised to 0 in GameObject.java
+
+    /**
+     * Tests the create methods - createPlayer() and createNPC().
+     */
+    @Test
+    void objectTest(){
+        System.out.println();
```
Comment (iluwatar) [inline component/src/test/java/com/iluwatar/component/GameObjectTest.java:24]: Use logger for output

File: component/src/test/java/com/iluwatar/component/GameObjectTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * Tests GameObject class.
+ * src/main/java/com/iluwatar/component/GameObject.java
+ */
+class GameObjectTest {
+
+    //creates player & npc objects for testing
+    GameObject playerTest = GameObject.createPlayer();
+    GameObject npcTest = GameObject.createNpc();
```
Comment (iluwatar) [inline component/src/test/java/com/iluwatar/component/GameObjectTest.java:15]: Maybe create these objects in @Before or @BeforeClass annotated method?

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Comment by LeseChe** [comment]
Thank you for the review, I will correct this in a few days.

File: component/src/test/java/com/iluwatar/component/GameObjectTest.java (hunk 3 above)
Comment (LeseChe) [reply component/src/test/java/com/iluwatar/component/GameObjectTest.java:15]: I notice that the @Before is renamed to @BeforeEach in juint5, just about to finish all changes. 🥇

**Comment by LeseChe** [comment]
@iluwatar  We have taken on your review feedback and have made appropriate changes. Please let me know if there is anything else that needs to be amended.

**Review by LeseChe — COMMENTED** [review]
Correct all the request changes.

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Good job! Well done for completing the new pattern implementation. Thank you for the contribution 🎉

@all-contributors please add @LeseChe for code

**Comment by iluwatar** [comment]
@all-contributors please add @LeseChe for code

**Pull request merged**